Kernel/Syscalls/profiling.cpp

```diff
@@ -19,12 +19,17 @@ u64 g_profiling_event_mask;
 // NOTE: event_mask needs to be passed as a pointer as u64
 //       does not fit into a register on 32bit architectures.
 ErrorOr<FlatPtr> Process::sys$profiling_enable(pid_t pid, Userspace<u64 const*> userspace_event_mask)
+{
```

Comment:
The syscall should retain the promises check.

Reply:
The promise is still checked in the call to Process::profiling_enable(), but there's no harm in duplicating the check.

Reply:
The point is that it doesn't make sense to check promises when this is being called from kernel mode already. So only the syscall should retain the promise check.

Reply:
Ohh, that makes more sense. I completely misunderstood your original comment. Should be fixed now.